### PR TITLE
Added support for Kerberos Authentication.

### PR DESCRIPTION
**Pull request by fang64**
I needed this for our environment, I believe I got it.

### CODE REVIEW

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -1661,6 +1663,25 @@
     end
   end
 
+  context 'with sshd_kerberos_authentication set to invalid value on valid osfamily' do
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: Apologies for the spec testing code here.. it's a mess. Would you mind adding a different describe section something like this

``` ruby
describe 'with KerberosAuthentication' do
  ['yes','no'].each do |value|
    context "set to #{value}" do
      # set param to value
      it { should contain_file('sshd_config').with_content(/^KerberosAuthentication #{value}$/) }
    end
  # put this new code where you set it to invalid here
  end
end
```

**Comment by ghoneycutt** [comment]
Seriously awesome work!!

Normally not a big deal, but since you'll need to rebase for the small spec test tweak, could you change the wording of the commit message? They should sound like a command, so "Add support for KerberosAuthentication option in sshd."

**Comment by fang64** [comment]
@ghoneycutt Alright I'll do that.

File: spec/classes/init_spec.rb (hunk 1 above)
Comment (fang64) [inline spec/classes/init_spec.rb]: so similar to this?
```puppet
describe 'with KerberosAuthentication' do
  ['yes','no','invalid'].each do |value|
    context "set to #{value}" do
      # set param to value
      it { should contain_file('sshd_config').with_content(/^KerberosAuthentication #{value}$/) }
    end
  end
end
```

I'm a little lost or don't bother with the invalid check?

**Comment by ghoneycutt** [comment]
I went ahead and fixed up your commit in PR #127 

Thanks a ton for adding this feature!

**Pull request closed without merging**